src/services/utility.service.ts

```diff
@@ -64,17 +89,25 @@ class UtilityService {
   public updateSearchResults = (
     inputText: string,
     setSearchResults: (locations: SearchResult[]) => void,
-    setLocationDisplay: (display: string) => void
+    setLocationDisplay: (display: string) => void,
+    currentLocation: Location
   ) => {
-    const locations: SearchResult[] = this.locationsToSearch.filter(
-      (location) => {
-        return (
-          location.searchName.toLowerCase().search(inputText.toLowerCase()) !==
-          -1
-        );
-      }
-    );
+    let locations: SearchResult[] = [];
+    locations = this.locationsToSearch.filter((location) => {
+      return (
+        location.searchName.toLowerCase().search(inputText.toLowerCase()) !== -1
+      );
+    });
 
+    if (currentLocation) {
+      const currentMarkerLocation: MarkerLocation = {
+        id: CURRENT_LOCATION_DISPLAY_TEXT,
+        displayName: CURRENT_LOCATION_DISPLAY_TEXT,
+        location: currentLocation,
+      };
+      const SearchResultMarker = MarkerToSearchResult(currentMarkerLocation);
+      locations.unshift(SearchResultMarker);
+    }
```

Comment:
Actually to make this simpler, you can change MarkerToSearchResult to take in a Location and return a SearchResult, which would save you from having to convert the currentLocation to a MarkerLocation.